packages/nx/src/project-graph/nx-deps-cache.ts

```diff
@@ -445,3 +456,12 @@ function getNxJsonPluginsData(
     };
   });
 }
+
+function getExternalNodesData(
```

Comment:
Do we have to reduce this or can we hash the whole thing?

Reply:
I think we should. Benchmarking both approaches in this repo:

```bash
Hash full external nodes: 27.798042000000002ms
Reduce and hash external nodes: 4.348999999999997ms
```

That's a ~6x time reduction, which is significant.